feat(deps): add optional kedro[pydantic] dependency extra (#5498)
kedro-org/kedro: RELEASE.md

* Added inspection API to get project snapshot.

## Bug fixes and other changes

> * Added `pydantic` optional dependency extra. Users can now install Pydantic support via `pip install "kedro[pydantic]"`.

Reviewer: ```suggestion
* Added an optional `pydantic` dependency extra, allowing users to enable Pydantic support with `pip install "kedro[pydantic]"`.
```

Reviewer: Hi @rahulbansod519 , can you please fix this ?
